### PR TITLE
DDCNL-6649: child benefit tile. Feature 1

### PR DESCRIPTION
**Pull request by SandeepUmredkar**

### CODE REVIEW

File: app/config/ConfigDecorator.scala [hunk 1]
```diff
@@ -223,6 +225,16 @@ class ConfigDecorator @Inject() (
...
+  lazy val claimChildBenefits: String = "https://www.gov.uk/child-benefit/how-to-claim"
+
+  lazy val reportChangesChildBenefit: String = "https://www.gov.uk/report-changes-child-benefit"
+
+  lazy val viewPaymentHistory: String = "https://www.gov.uk/child-benefit/view-payment-history"
+
+  lazy val viewProofEntitlement: String = "https://www.gov.uk/child-benefit/view-proof-entitlement"
+
+  lazy val childBenefitTaxCharge: String = "https://www.gov.uk/child-benefit-tax-charge"
+
```
Comment (pascal22p) [inline app/config/ConfigDecorator.scala:237]: We should link to the welsh version of those urls if they exist if welsh is used on our side.

File: app/controllers/InterstitialController.scala [hunk 2]
```diff
@@ -96,12 +98,32 @@ class InterstitialController @Inject() (
     configDecorator.pertaxFrontendHost + request.path
 
   def displayChildBenefits: Action[AnyContent] = authenticate { implicit request =>
-    Ok(
-      viewChildBenefitsSummaryInterstitialView(
-        redirectUrl = currentUrl,
-        taxCreditsEnabled = configDecorator.taxCreditsEnabled
+    val childBenefitSingleAccountToggleValue: Option[Try[FeatureFlag]] =
+      featureFlagService.get(ChildBenefitSingleAccountToggle).value
+    if (childBenefitSingleAccountToggleValue.isEmpty || !childBenefitSingleAccountToggleValue.get.get.isEnabled) {
```
Comment (pascal22p) [inline app/controllers/InterstitialController.scala:103]: `.get.get` ?? Use a map:

`
    featureFlagService.get(ChildBenefitSingleAccountToggle).map { featureFlag =>
      if (featureFlag.isEnabled) {
        ???
      } else ???
   }
`

File: app/controllers/InterstitialController.scala [hunk 3]
```diff
@@ -96,12 +98,32 @@ class InterstitialController @Inject() (
...
+      )
+    } else {
+      errorRenderer.error(UNAUTHORIZED)
+    }
+  }
+
+  def displayChildBenefitsSingleAccountView: Action[AnyContent] = authenticate { implicit request =>
+    val childBenefitSingleAccountToggleValue: Option[Try[FeatureFlag]] =
+      featureFlagService.get(ChildBenefitSingleAccountToggle).value
+    if (childBenefitSingleAccountToggleValue.nonEmpty && childBenefitSingleAccountToggleValue.get.get.isEnabled) {
```
Comment (pascal22p) [inline app/controllers/InterstitialController.scala:118]: Same as above

File: app/controllers/controllershelpers/HomeCardGenerator.scala [hunk 4]
```diff
@@ -159,8 +157,15 @@ class HomeCardGenerator @Inject() (
   def getTaxCreditsCard(showTaxCreditsPaymentLink: Boolean)(implicit messages: Messages): Some[HtmlFormat.Appendable] =
     Some(taxCreditsView(showTaxCreditsPaymentLink))
 
-  def getChildBenefitCard()(implicit messages: Messages): Some[HtmlFormat.Appendable] =
-    Some(childBenefitView())
+  def getChildBenefitCard()(implicit messages: Messages): Option[HtmlFormat.Appendable] = {
+    val childBenefitSingleAccountToggleValue: Option[Try[FeatureFlag]] =
+      featureFlagService.get(ChildBenefitSingleAccountToggle).value
+    if (childBenefitSingleAccountToggleValue.nonEmpty && childBenefitSingleAccountToggleValue.get.get.isEnabled) {
```
Comment (pascal22p) [inline app/controllers/controllershelpers/HomeCardGenerator.scala:163]: Again, use map

**Review by pascal22p — CHANGES_REQUESTED** [review]
see comments

File: app/controllers/InterstitialController.scala [hunk 5]
```diff
@@ -95,13 +96,33 @@ class InterstitialController @Inject() (
...
+    featureFlagService.get(ChildBenefitSingleAccountToggle).map { featureFlag =>
+      if (!featureFlag.isEnabled) {
+        Ok(
+          viewChildBenefitsSummaryInterstitialView(
+            redirectUrl = currentUrl,
+            taxCreditsEnabled = configDecorator.taxCreditsEnabled
+          )
+        )
+      } else {
+        errorRenderer.error(UNAUTHORIZED)
```
Comment (pascal22p) [inline app/controllers/InterstitialController.scala:109]: Can you redirect to the new page using a 301 status.
Remember that urls get bookmarked or can be used by other services. Always send the user to the new location is the url changes.

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: app/config/ConfigDecorator.scala (hunk 1 above)
Comment (pascal22p) [reply app/config/ConfigDecorator.scala]: This has been taken to gov.uk for the welsh urls

**Review by pascal22p — APPROVED** [review]

File: app/config/ConfigDecorator.scala [hunk 6]
```diff
@@ -221,7 +223,22 @@ class ConfigDecorator @Inject() (
...
+  lazy val claimChildBenefits: String = "https://www.gov.uk/child-benefit/how-to-claim"
+
+  lazy val reportChangesChildBenefit: String = "https://www.gov.uk/report-changes-child-benefit"
+
+  lazy val reportChangesChildBenefitWelsh: String = "https://www.gov.uk/rhoi-gwybod-am-newidiadau-budd-dal-plant"
+
+  lazy val viewPaymentHistory: String = "https://www.gov.uk/child-benefit/view-payment-history"
+
+  lazy val viewProofEntitlement: String = "https://www.gov.uk/child-benefit/view-proof-entitlement"
+
```
Comment (pascal22p) [inline app/config/ConfigDecorator.scala:238]: We need the welsh

File: app/config/ConfigDecorator.scala [hunk 7]
```diff
@@ -221,7 +223,22 @@ class ConfigDecorator @Inject() (
     s"$dfsFrontendHost/digital-forms/form/authorise-a-tax-adviser-for-high-income-child-benefit-charge-matters/draft/guide"
   lazy val childBenefitsStopOrRestart               =
     s"$dfsFrontendHost/digital-forms/form/high-income-child-benefit-tax-charge/draft/guide"
-  lazy val childBenefitsCheckIfYouCanClaim          = "https://www.gov.uk/child-benefit/overview"
+
+  lazy val childBenefitsCheckIfYouCanClaim: String = "https://www.gov.uk/child-benefit/overview"
+
+  lazy val claimChildBenefits: String = "https://www.gov.uk/child-benefit/how-to-claim"
+
```
Comment (pascal22p) [inline app/config/ConfigDecorator.scala:230]: We need the welsh

File: app/config/ConfigDecorator.scala [hunk 8]
```diff
@@ -221,7 +223,22 @@ class ConfigDecorator @Inject() (
...
+  lazy val reportChangesChildBenefit: String = "https://www.gov.uk/report-changes-child-benefit"
+
+  lazy val reportChangesChildBenefitWelsh: String = "https://www.gov.uk/rhoi-gwybod-am-newidiadau-budd-dal-plant"
+
+  lazy val viewPaymentHistory: String = "https://www.gov.uk/child-benefit/view-payment-history"
+
```
Comment (pascal22p) [inline app/config/ConfigDecorator.scala:236]: We need the welsh

**Review by pascal22p — CHANGES_REQUESTED** [review]

**Review by pascal22p — DISMISSED** [review]

**Review by askar-makkunnath — APPROVED** [review]

**Pull request merged**